src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupLoggingHelper.cs

```diff
@@ -16,18 +16,15 @@ namespace Microsoft.Build.BackEnd
     /// </summary>
     internal static class ItemGroupLoggingHelper
     {
-        /// <summary>
-        /// Gets a text serialized value of a parameter for logging.
-        /// </summary>
-        internal static string GetParameterText(string prefix, string parameterName, params object[] parameterValues)
-        {
-            return GetParameterText(prefix, parameterName, (IList)parameterValues);
-        }
+        internal static string ItemGroupIncludeLogMessagePrefix = ResourceUtilities.GetResourceString("ItemGroupIncludeLogMessagePrefix");
```

Comment:
Rather than caching four specific strings (presumably the most oft-used?), could you initialize an empty dictionary and cache strings in there as you need them?

Reply:
Actually, these are the only strings that are used in normal conditions, right? So this seems reasonable.

Reply:
Yes, the strings are fixed always.

There wasn't an OrderBy but I was considering various approaches to sort, and decided to measure properly. Good thing I didn't go with it ;) https://twitter.com/KirillOsenkov/status/1245180011457306624

Reply:
Is there anything wrong with an unstable sort here? Metadata all have unique names.

Reply:
We should be good here. I doubt they really have to be sorted in the first place. The viewer could sort when loading the binlog. But I guess there's some merit to it if they always appear in the same order, maybe some log parsing tools rely on that. 

Although if a log parsing tool isn't using binlog they just deserve the pain they get ;)

Reply:
I see an argument both ways, since having a consistent order is good, but so is allocating less. I'm tempted to put out a no-sort version under a feature flag and see what people think, but it probably isn't worth it.

Reply:
Yes, there's just so much more low-hanging fruit everywhere. I'd love to fix that first.

Reply:
We should take note of other places where we can cache resource strings like this 👍